### PR TITLE
azurerm_function_app - support for cors

### PR DESCRIPTION
**Pull request by hbuckle**
Another copy and paste job from the app service resource to support cors for functions

### CODE REVIEW

File: azurerm/resource_arm_function_app.go [hunk 1]
```diff
@@ -199,6 +199,26 @@ func resourceArmFunctionApp() *schema.Resource {
 							Optional: true,
 							Computed: true,
 						},
+						"cors": {
```
Comment (katbyte) [inline azurerm/resource_arm_function_app.go]: If this is the same schema used in `app_service` could we maybe create a shares schema function here?

File: azurerm/resource_arm_function_app.go [hunk 2]
```diff
@@ -681,6 +701,12 @@ func expandFunctionAppSiteConfig(d *schema.ResourceData) web.SiteConfig {
 		siteConfig.LinuxFxVersion = utils.String(v.(string))
 	}
 
+	if v, ok := config["cors"]; ok {
+		corsSettings := v.(interface{})
+		expand := azure.ExpandAppServiceCorsSettings(corsSettings)
```
Comment (katbyte) [inline azurerm/resource_arm_function_app.go]: As this is now used by two resources might be more aptly named:
```suggestion
		expand := azure.ExpandWebCorsSettings(corsSettings)
```

File: azurerm/resource_arm_function_app.go [hunk 3]
```diff
@@ -709,6 +735,8 @@ func flattenFunctionAppSiteConfig(input *web.SiteConfig) []interface{} {
 		result["linux_fx_version"] = *input.LinuxFxVersion
 	}
 
+	result["cors"] = azure.FlattenAppServiceCorsSettings(input.Cors)
```
Comment (katbyte) [inline azurerm/resource_arm_function_app.go]: As above

**Comment by hbuckle** [comment]
@katbyte - I've moved the relevant bits into a new web.go helper

**Pull request merged**